src/app/fdctl/run/tiles/metrics.c

```diff
@@ -0,0 +1,392 @@
+#include "../../fdctl.h"
+#include "../run.h"
+
+#include "../../../../disco/fd_disco.h"
+#include "../../../../util/metrics/fd_metrics.h"
+#include "../../../../util/metrics/fd_metrics_impl.h"
+#include <stdio.h>
+#include <errno.h>
+
+#include <linux/unistd.h>
+#include <arpa/inet.h>
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+#define __USE_MISC
+#include <netdb.h>
+
+#define DEBUG 0
+
+#define METRICS_TAG 0x8bf72a57ca2947b7UL
+
+static void
+ssl_init( void ) {
+  SSL_library_init();
+  OpenSSL_add_all_algorithms();
+  SSL_load_error_strings();
+  ERR_load_BIO_strings();
+  ERR_load_crypto_strings();
+}
+
+ulong
+append_datapoints( char *       append_buffer,
+                   ulong        append_buffer_capacity,
+                   char * const tile_name,
+                   ulong        idx,
+                   Datapoint *  datapoints,
+                   ulong        datapoints_sz ) {
+  metrics_tag_t tags[] = {
+      { .key   = "host_id",
+         .value = "" },     /* strncpy below from fd_log_host() */
+      { .key   = "tile",
+          .value = "" },    /* snprintf below */
+      { .key   = "index",
+          .value = "" },    /* snprintf below */
+  };
+  strncpy(  tags[ 0 ].value, fd_log_host(), sizeof( tags[ 0 ].value ) - 1 );
+  strncpy(  tags[ 1 ].value, tile_name,     sizeof( tags[ 1 ].value ) - 1 );
+  snprintf( tags[ 2 ].value, sizeof( tags[ 2 ].value ), "%ld", idx );
+
+  long ts = fd_log_wallclock();
+  for( ulong i=0; i<datapoints_sz; i++ ) {
+    char data[ 4096 ];
+    int ret = fd_metrics_format( data, sizeof( data ), tile_name, tags, sizeof( tags )/sizeof( tags[0] ), &datapoints[i], 1, ts++ );
+    if( FD_UNLIKELY( ret<0 || ret>=(int)sizeof( data ) ) ) {
+      FD_LOG_WARNING(( "snprintf error %d", ret ));
+      return 0;
+    }
+    ulong line_sz = strlen( data ) + 1;
+    if( FD_UNLIKELY( append_buffer_capacity <= line_sz ) ) {
+      FD_LOG_WARNING(( "append_buffer_capacity %lu <= %lu", append_buffer_capacity, strlen( data ) + 1 ));
+      return 0;
+    }
+    strncat( append_buffer, data, append_buffer_capacity - 1 );
+    append_buffer_capacity -= line_sz;
+  }
+  return append_buffer_capacity;
+}
+
+void
+ssl_send_data( const char * _server,
+               ushort port,
+               const char * user_pass,
+               const char * db,
+               char * post_buffer,
+               const ulong post_buffer_sz,
+               char * const data ) {
+  int ssl_sockfd = socket( AF_INET, SOCK_STREAM, 0 );
+  if( FD_UNLIKELY( ssl_sockfd < 0 ) ) {
+    FD_LOG_WARNING(( "socket error %d (%d) %s", ssl_sockfd, errno, strerror( errno ) ));
+    return;
+  }
+
+  /* Resolve the server address */
+  struct hostent * server = gethostbyname( _server );
```

Comment:
Per discussion, kind of hate this glibc function, god knows what it ends up doing. Could we just include the MUSL `gethostbyname` code directly?

Reply:
I also think gethostbyname is blocking, which has the same issues as below. This could be async / polled as part of the run loop instead.